conda_forge_tick/update_upstream_versions.py

```diff
@@ -9,11 +9,63 @@
 from conda.models.version import VersionOrder
 from pkg_resources import parse_version
 
-from .utils import parse_meta_yaml, setup_logger
+from utils import parse_meta_yaml, setup_logger
+import os
 
 logger = logging.getLogger("conda_forge_tick.update_upstream_versions")
 
 
+class LibrariesIO:
+
+    def __init__(self):
+        self.libraries_io_api_key = os.getenv("LIBRARIES_IO_API_KEY")
+        self._data = None
+
+    def get_libraries_io_subscriptions(self):
+        if self.libraries_io_api_key == None:
+            return
+        l = []
+        for i in range(1, 2):
+            url = "https://libraries.io/api/subscriptions?api_key={}&page={}&per_page={}&include_prerelease=False"
+            url = url.format(self.libraries_io_api_key, i, 100)
+            print(url.replace(self.libraries_io_api_key, "<dummy>"))
+            r = retry_requests(requests.get, url)
+            if not r.ok or len(r.json()) == 0:
+                break
+            l.extend(r.json())
+        return l
+
+    @property
+    def data(self):
+        if self._data is not None:
+            return self._data
+        d = {'pypi':{}, 'cran':{}}
+        for proj in self.get_libraries_io_subscriptions():
+            platform = proj['project']['platform'].lower()
+            d[platform][proj['project']['name']] = proj
+        self._data = d
+        return d
+
+    def subscribe(self, platform, pkg):
+        post_url = "https://libraries.io/api/subscriptions/{}/{}?api_key={}&include_prerelease=false"
+        post_url = post_url.format(platform, pkg, self.libraries_io_api_key)
+        r = retry_requests(requests.post, post_url)
+        c = r.json()
+        platform = c['project']['platform'].lower()
+        self.data[platform][c['project']['name']] = c
+        return c
+
+    def get_package_versions(self, platform, pkg):
+        if pkg in self.data[platform]:
+            d = self.data[platform][pkg]
+        else:
+            d = self.subscribe(platform, pkg)
+        return [ver['number'] for ver in d['project']['versions']]
```

Comment:
The json has a key `latest_release_version`. Should that be used instead?

Reply:
I haven't checked yet. Does it remove rc, beta, dev versions?